uint64_t': ambiguous symbol error in Benchmarks_test.cc under MSVC 2015

Error	C2872	'uint64_t': ambiguous symbol	benchmarks_test	entityx\entityx\Benchmarks_test.cc	51	
this error also appears for lines 54, 61 and 63

Manually adding an include to stdint.h doesn't seem to fix it.

I've reproduced and found the ambiguous symbol error:

```
2>C:\Users\Stephen\OneDrive\Projects\github\entityx\entityx\Benchmarks_test.cc(47): error C2872: 'uint64_t': ambiguous symbol
2>  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\stdint.h(24): note: could be 'unsigned __int64 uint64_t'
2>  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\stdint.h(24): note: or       'uint64_t'
2>  C:\Users\Stephen\OneDrive\Projects\github\entityx\entityx/Entity.h(40): note: or       'entityx::uint64_t'

```

It seems the Benchamark tests uses both the std:: and entityx:: namespaces. MSVC can't decide dispite the `using std::uint64_t;` in the benchmark_test.cc.

It might be a compiler difference between GCC and MSVC? Very curious why the `using std;:uint64_t` didn't work.
 
A workaround is to remove `using namespace std` and `using namespace entityx`, and add this in:

```
using std::uint64_t;
using std::cout;
using std::endl;
using std::vector;

using entityx::Receiver;
using entityx::Component;
using entityx::ComponentHandle;
using entityx::EntityCreatedEvent;
using entityx::EntityDestroyedEvent;
using entityx::EventManager;
using entityx::EntityManager;
using entityx::Entity;
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,20 +28,20 @@ include(CheckCXXSourceCompiles)
 
 # Default compiler args
 if (CMAKE_CXX_COMPILER_ID MATCHES "(GNU|.*Clang)")
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Werror -Wall -Wextra -Wno-unused-parameter -Wno-error=unused-variable -Wno-error=sign-compare -std=c++11")
-	set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g")
-	set(CMAKE_CXX_FLAGS_MINSIZEREL "-g -Os -DNDEBUG")
-	set(CMAKE_CXX_FLAGS_RELEASE "-g -O2 -DNDEBUG")
-	set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Werror -Wall -Wextra -Wno-unused-parameter -Wno-error=unused-variable -Wno-error=sign-compare -std=c++11")
+    set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g")
+    set(CMAKE_CXX_FLAGS_MINSIZEREL "-g -Os -DNDEBUG")
+    set(CMAKE_CXX_FLAGS_RELEASE "-g -O2 -DNDEBUG")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL 'MSVC')
-	# /Zi - Produces a program database (PDB) that contains type information and symbolic debugging information for use with the debugger.
-	# /FS - Allows multiple cl.exe processes to write to the same .pdb file
-	# /DEBUG - Enable debug during linking
-	# /Od - Disables optimization
-	set(CMAKE_CXX_FLAGS_DEBUG "/Zi /FS /DEBUG /Od /MDd")
-	# /Ox - Full optimization
-	set(CMAKE_CXX_FLAGS_RELEASE "/Ox -DNDEBUG")
-	set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/Ox /Zi /FS /DEBUG")
+    # /Zi - Produces a program database (PDB) that contains type information and symbolic debugging information for use with the debugger.
+    # /FS - Allows multiple cl.exe processes to write to the same .pdb file
+    # /DEBUG - Enable debug during linking
+    # /Od - Disables optimization
+    set(CMAKE_CXX_FLAGS_DEBUG "/Zi /FS /DEBUG /Od /MDd")
+    # /Ox - Full optimization
+    set(CMAKE_CXX_FLAGS_RELEASE "/Ox -DNDEBUG")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/Ox /Zi /FS /DEBUG")
 endif()
 
 # if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
@@ -59,7 +59,7 @@ include(CheckCXX11Features.cmake)
 
 set(OLD_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 if ((MAKE_CXX_COMPILER_ID STREQUAL "GNU") OR (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
 endif()
 check_cxx_source_compiles(
 "
@@ -99,7 +99,15 @@ macro(create_test TARGET_NAME SOURCE)
         ${ARGN}
         )
     set_target_properties(${TARGET_NAME} PROPERTIES FOLDER "entityx/tests")
-    add_test(${TARGET_NAME} ${TARGET_NAME})
+
+    # Special case for benchmark tests
+    if (${TARGET_NAME} MATCHES .*benchmark.*)
+        if(ENTITYX_RUN_BENCHMARKS)
+            add_test(${TARGET_NAME} ${TARGET_NAME})
+        endif()
+    else()
+        add_test(${TARGET_NAME} ${TARGET_NAME})
+    endif()
 endmacro()
 
 if (NOT CMAKE_BUILD_TYPE)
@@ -150,9 +158,9 @@ if (ENTITYX_BUILD_TESTING)
     create_test(system_test entityx/System_test.cc)
     create_test(tags_component_test entityx/tags/TagsComponent_test.cc)
     create_test(dependencies_test entityx/deps/Dependencies_test.cc)
+    create_test(benchmarks_test entityx/Benchmarks_test.cc)
     if (ENTITYX_RUN_BENCHMARKS)
         message("-- Running benchmarks")
-        create_test(benchmarks_test entityx/Benchmarks_test.cc)
     else ()
         message("-- Not running benchmarks (use -DENTITYX_RUN_BENCHMARKS=1 to enable)")
     endif ()

--- a/entityx/Benchmarks_test.cc
+++ b/entityx/Benchmarks_test.cc
@@ -6,11 +6,19 @@
 #include "entityx/help/Timer.h"
 #include "entityx/Entity.h"
 
-using namespace std;
-using namespace entityx;
-
 using std::uint64_t;
+using std::cout;
+using std::endl;
+using std::vector;
 
+using entityx::Receiver;
+using entityx::Component;
+using entityx::ComponentHandle;
+using entityx::Entity;
+using entityx::EntityCreatedEvent;
+using entityx::EntityDestroyedEvent;
+using entityx::EventManager;
+using entityx::EntityManager;
 
 struct AutoTimer {
   ~AutoTimer() {

